### PR TITLE
SAP-6788 Fix pedigree view with children as horizontal edge nodes

### PR DESCRIPTION
**Pull request by ybaryshnikova**

### CODE REVIEW

File: src/xcoordclass.js [hunk 1]
```diff
@@ -129,6 +133,10 @@ XCoord.prototype = {
         // as soon as it get as close as allowed to it's right neighbour
 
     var rightBoundary = this.getRightMostNoDisturbPosition(v);
+    if (rightBoundary === Infinity || rightBoundary === -Infinity) {
```
Comment (Paca-Vaca) [inline src/xcoordclass.js]: Isn't it the code written twice? 😃

Comment (Paca-Vaca) [reply src/xcoordclass.js]: ...with one variable

Comment (Paca-Vaca) [reply src/xcoordclass.js]: `const returnIfOutside = boundary => (boundary === (Infinity || boundary === -Infinity) && return 0)`

Comment (Chris-Congenica) [reply src/xcoordclass.js]: At the very least we could probably use `isFinite` to shorten this a little.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite

`if (Number.isFinite(rightBoundary)) return 0;`

Comment (Paca-Vaca) [reply src/xcoordclass.js]: Just a question about the coding habits - does somebody prefer to use where it's possible such forms?: `(rightBoundary === Infinity || rightBoundary === -Infinity) && return 0`

Comment (Paca-Vaca) [reply src/xcoordclass.js]: ...or even `Number.isFinite(rightBoundary) && return 0`?

Comment (ybaryshnikova) [reply src/xcoordclass.js]: simplified the check using `Number.isFinite` and moved it to a function. As to the notation, personally, I prefer using `if` over a short form, it seems a bit more readable to me. Let's see what other reviewers think?

Comment (Chris-Congenica) [reply src/xcoordclass.js]: I generally prefer the if format it is more obvious when used outside of React.

**Review by Chris-Congenica — APPROVED** [review]

**Pull request merged**